lib/compile.js

```diff
@@ -0,0 +1,155 @@
+const { normalize, resolve } = require('path');
+const { tags, parsers, converters } = require('./statements');
+
+
+const getDir = (path) => path.replace(/\/[^\/]+$/, '');
+const cleanWhitespace = (str) => str.replace(/\s+/g, ' ').trim();
+const escape = (str) => str.replace(/`/gm, '\\\`');
+
+
+function resolvePath(path, parentPath, root) {
+  return path.startsWith('/')
+    ? path
+    : path.startsWith('~')
+      ? resolve(root, path.replace(/^~/, '.'))
+      : normalize(`${getDir(parentPath)}/${path}`);
+}
+
+
+class BeardError {
```

Comment:
@shanebo Is this beard error ever useful to you? I'm wondering if we should just drop it now, unless it is actually useful to you.

Reply:
@joeosburn yeah it tells me when it's a beard syntax problem vs something else.

Reply:
👍